app/repositories/LockRepository.scala

```diff
@@ -16,86 +16,71 @@
 
 package repositories
 
+import com.google.inject.ImplementedBy
+import com.mongodb.MongoWriteException
 import config.AppConfig
 import models.DepartureId
-import models.MongoDateTimeFormats._
-import play.api.libs.json.Json
-import play.modules.reactivemongo.ReactiveMongoApi
-import reactivemongo.api.WriteConcern
-import reactivemongo.api.bson.BSONDocument
-import reactivemongo.api.bson.collection.BSONSerializationPack
-import reactivemongo.api.commands.LastError
-import reactivemongo.api.indexes.Index.Aux
-import reactivemongo.api.indexes.IndexType
-import reactivemongo.play.json.collection.Helpers.idWrites
-import reactivemongo.play.json.collection.JSONCollection
-import utils.IndexUtils
+import models.DepartureLock
+import org.mongodb.scala.model.Filters
+import org.mongodb.scala.model.IndexModel
+import org.mongodb.scala.model.IndexOptions
+import org.mongodb.scala.model.Indexes
+import uk.gov.hmrc.mongo.MongoComponent
+import uk.gov.hmrc.mongo.play.json.Codecs
+import uk.gov.hmrc.mongo.play.json.PlayMongoRepository
 
 import java.time.LocalDateTime
+import java.util.concurrent.TimeUnit
 import javax.inject.Inject
+import javax.inject.Singleton
 import scala.concurrent.ExecutionContext
 import scala.concurrent.Future
 
-class LockRepository @Inject()(mongo: ReactiveMongoApi, appConfig: AppConfig)(implicit ec: ExecutionContext) {
-
-  private val documentExistsErrorCodeValue = 11000
+@ImplementedBy(classOf[LockRepositoryImpl])
+trait LockRepository {
```

Comment:
I imagine this is locking departure writes. HMRC mongo has its own lock service, can we use that instead?

https://github.com/hmrc/hmrc-mongo#lockservice

Reply:
if you can see this LockRepository is using simple locking mechanism (i.e. using lock and unlock method) and not any internal locking mechanism so I found that it is the easiest way of doing this.  If you can see on migration guide https://github.com/hmrc/hmrc-mongo/blob/main/MIGRATION.md#lock that is the locking mechanism used in reactive mongo but we are not using that so there is no need to migrate that with hmrc locking mechanism. As if we try to migrate that with hmrc lock then going to be totally new approach  and we need to modify the service and action classes as well because lock and unlock method are used there and we need to change that logic as well.